### PR TITLE
urh: 2.9.6 -> 2.9.8; install desktop file

### PR DESCRIPTION
**Pull request by pawsen**
This PR does three things

- Update `urh` to newest release.
- fix: prevent double wrapping of the binary, introduced in #357400. 
  See https://nixos.org/manual/nixpkgs/stable/#ssec-gnome-common-issues-double-wrapped
- install a `urh.desktop` file and re-scale the logo to `48x48x` and `128x128`

## Things done

- Built on platform(s)
  - [x ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ x ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ur/urh/package.nix [hunk 1]
```diff
@@ -25,12 +30,36 @@ python3Packages.buildPythonApplication rec {
...
 
   doCheck = false;
 
+  desktopItems = [
+    (makeDesktopItem {
+      name = pname;
+      exec = "urh";
+      icon = "urh";
+      desktopName = "Universal Radio Hacker";
+      categories = [ "HamRadio" ];
```
Comment (fpletz) [inline pkgs/by-name/ur/urh/package.nix:52]:
```suggestion
      categories = [ "Network" "HamRadio" ];
```
As an additional category, HamRadio should also have a main category. I would suggest Network, see https://specifications.freedesktop.org/menu-spec/latest/additional-category-registry.html.

Comment (pawsen) [reply pkgs/by-name/ur/urh/package.nix]: Thanks, appreciated!

File: pkgs/by-name/ur/urh/package.nix [hunk 2]
```diff
@@ -25,12 +30,36 @@ python3Packages.buildPythonApplication rec {
...
+      desktopName = "Universal Radio Hacker";
+      categories = [ "Network" "HamRadio" ];
+      comment = meta.description;
+    })
+  ];
+
+  postInstall = ''
+    mkdir -p $out/share/icons/hicolor/{48x48,128x128}/apps
+    convert $src/data/icons/appicon.png -resize 48x48 $out/share/icons/hicolor/48x48/apps/urh.png
+    convert $src/data/icons/appicon.png -resize 128x128 $out/share/icons/hicolor/128x128/apps/urh.png
```
[the remark this answers is not in the document]
Comment (pawsen) [reply pkgs/by-name/ur/urh/package.nix]: Thanks. I was unsure if the icon size was too big(512x512).
This is much neater.

File: pkgs/by-name/ur/urh/package.nix [hunk 3]
```diff
@@ -1,22 +1,26 @@
...
   pname = "urh";
-  version = "2.9.6";
+  version = "2.9.8";
 
   src = fetchFromGitHub {
     owner = "jopohl";
     repo = pname;
     rev = "refs/tags/v${version}";
-    sha256 = "sha256-4Fe2+BUdnVdNQHqZeftXLabn/vTzgyynOtqy0rAb0Rk=";
+    sha256 = "sha256-r3d80dzGwgf5Tuwt1IWGcmNbblwBNKTKKm+GGx1r2HE=";
```
Comment (Aleksanaa) [inline pkgs/by-name/ur/urh/package.nix:19]:
```suggestion
    hash = "sha256-r3d80dzGwgf5Tuwt1IWGcmNbblwBNKTKKm+GGx1r2HE=";
```

File: pkgs/by-name/ur/urh/package.nix [hunk 4]
```diff
@@ -1,22 +1,26 @@
...
     owner = "jopohl";
     repo = pname;
     rev = "refs/tags/v${version}";
-    sha256 = "sha256-4Fe2+BUdnVdNQHqZeftXLabn/vTzgyynOtqy0rAb0Rk=";
+    hash = "sha256-r3d80dzGwgf5Tuwt1IWGcmNbblwBNKTKKm+GGx1r2HE=";
   };
 
-  nativeBuildInputs = [ qt5.wrapQtAppsHook wrapGAppsHook3 ];
+  nativeBuildInputs = [ qt5.wrapQtAppsHook wrapGAppsHook3 copyDesktopItems
+                      ];
```
Comment (Aleksanaa) [inline pkgs/by-name/ur/urh/package.nix:23]:
```suggestion
  nativeBuildInputs = [ qt5.wrapQtAppsHook wrapGAppsHook3 copyDesktopItems ];
```

File: pkgs/by-name/ur/urh/package.nix [hunk 5]
```diff
@@ -25,12 +28,34 @@ python3Packages.buildPythonApplication rec {
...
+
   postFixup = ''
     wrapQtApp $out/bin/urh
   '';
 
   doCheck = false;
 
+  desktopItems = [
+    (makeDesktopItem {
+      name = pname;
```
Comment (Aleksanaa) [inline pkgs/by-name/ur/urh/package.nix:46]:
```suggestion
      name = "urh";
```

File: pkgs/by-name/ur/urh/package.nix [hunk 6]
```diff
@@ -25,12 +28,34 @@ python3Packages.buildPythonApplication rec {
...
+  # dont double wrap
+  # https://nixos.org/manual/nixpkgs/stable/#ssec-gnome-common-issues-double-wrapped
+  dontWrapGApps = true;
+  preFixup = ''
+    qtWrapperArgs+=("''${gappsWrapperArgs[@]}")
+  '';
+
   postFixup = ''
     wrapQtApp $out/bin/urh
   '';
```
Comment (Aleksanaa) [inline pkgs/by-name/ur/urh/package.nix:40]:
```suggestion
```

**Comment by Aleksanaa** [comment]
@ofborg build urh

File: pkgs/by-name/ur/urh/package.nix [hunk 7]
```diff
@@ -25,12 +28,30 @@ python3Packages.buildPythonApplication rec {
...
   ];
 
-  postFixup = ''
-    wrapQtApp $out/bin/urh
+  # dont double wrap
+  # https://nixos.org/manual/nixpkgs/stable/#ssec-gnome-common-issues-double-wrapped
+  dontWrapGApps = true;
+  preFixup = ''
+    qtWrapperArgs+=("''${gappsWrapperArgs[@]}")
   '';
```
Comment (Aleksanaa) [inline pkgs/by-name/ur/urh/package.nix:36]:
```suggestion
  dontWrapGApps = true;
  dontWrapQtApps = true;
  preFixup = ''
    makeWrapperArgs+=(
      ''${gappsWrapperArgs[@]}
      ''${qtWrapperArgs[@]}
    )
  '';
```

**Pull request merged**